include/gradual_flow/gcode_path.h

```diff
@@ -339,13 +351,24 @@ struct GCodeState
      */
     std::vector<GCodePath> processGcodePath(const GCodePath& path, const utils::Direction direction)
     {
+        if (flow_state == FlowState::UNDEFINED)
+        {
+            current_flow = setpoint_flow;
+        }
+
         if (path.isTravel())
         {
+            undefined_state_duration += path.totalDuration();
+            if (path.retract || undefined_state_duration > discretized_duration)
+            {
+                flow_state = FlowState::UNDEFINED;
+            }
             return { path };
         }
+        undefined_state_duration = 0;
```

Comment:
Since `undefined_state_duration` is always reset to 0 after the if-block it's not really required to store it on the struct

```suggestion
        if (path.isTravel())
        {
            if (path.retract || path.totalDuration() > discretized_duration)
            {
                flow_state = FlowState::UNDEFINED;
            }
            return { path };
        }
```

Reply:
Also I'm not really sure if we should use the `discretized_duration` here. I would not expect the "undefined state" behaviour to change if I were to increase/decrease the gradual flow behavior.

Reply:
Fair assessment let's make it a setting